### ISSUE TITLE
SPF: Two domains both referring to each other in includes.

### ISSUE THREAD

**Issue by JesseLukeWalker** [omniomi/PSMailTools#1]
"ValidateSPF : The script failed due to call depth overflow."
"Resolve-SPFRecord : The script failed due to call depth overflow."

As an example, there is tek-experts.com which has include:ynvholdings.com.cy in the SPF record, while ynvholdings.com.cy has include:tek-experts.com.

https://mxtoolbox.com/SuperTool.aspx?action=spf%3atek-experts.com

I'd assume that your script should be coded to allow the first include: expansion, but not go ahead with the second expansion to resolve this issue - i.e., don't expand any includes that mention the domain which is being queried, or at least flag them up and show a note or something.

**Comment by JesseLukeWalker** [comment]
I notice also that https://www.kitterman.com/spf/validate.html also errors out on this domain with the message "Results - PermError SPF Permanent Error: Too many DNS lookups", so it's not just you, but probably something that can be coded for nonetheless.

**Comment by omniomi** [comment]
Good catch. I ran the same domain through my bash and web based versions of the tool to equally broken results. I'll need to come up with some sort of loop prevention.
